packages/nimble-components/src/table-column/mixins/custom-sort-order.ts

```diff
@@ -0,0 +1,110 @@
+    TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>
+> = Pick<TableColumn<unknown, TColumnValidator>, 'columnInternals'>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type CustomSortOrderTableColumnConstructor<TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>> = abstract new (...args: any[]) => CustomSortOrderTableColumn<TColumnValidator>;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinCustomSortOrderColumnAPI<
+    TBase extends CustomSortOrderTableColumnConstructor<TColumnValidator>,
+    TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow sorting
+     * by a field other than the default for that column.
+     */
+    abstract class CustomSortOrderColumn extends base {
+        public sortByFieldName?: string;
+
+        /** @internal */
+        public customSortOrderColumnNotifier?: Notifier;
+
+        /** @internal */
+        public customSortOrderColumnChangeHandler = {
+            handleChange: this.handleCustomSortOrderColumnChange.bind(this)
+        };
+
+        public getResolvedOperandDataRecordFieldName(
+            initialOperandFieldName: string | undefined
+        ): string | undefined {
+            return typeof this.sortByFieldName === 'string'
+                ? this.sortByFieldName
+                : initialOperandFieldName;
+        }
+
+        public getResolvedSortOperation(
+            initialSortOperation: TableColumnSortOperation
+        ): TableColumnSortOperation {
+            return typeof this.sortByFieldName === 'string'
+                ? TableColumnSortOperation.basic
+                : initialSortOperation;
+        }
+
+        /** @internal */
+        public sortByFieldNameChanged(): void {
+            this.handleSortByFieldNameChange();
+            this.updateCustomColumnSortingValidity();
+
+            if (
+                typeof this.sortByFieldName === 'string'
+                && !this.customSortOrderColumnNotifier
+            ) {
+                this.customSortOrderColumnNotifier = Observable.getNotifier(
+                    this.columnInternals
+                );
+                this.customSortOrderColumnNotifier.subscribe(
+                    this.customSortOrderColumnChangeHandler
+                );
+            } else {
+                this.customSortOrderColumnNotifier?.unsubscribe(
+                    this.customSortOrderColumnChangeHandler
+                );
+                this.customSortOrderColumnNotifier = undefined;
+            }
+        }
+
+        public abstract handleSortByFieldNameChange(): void;
```

Comment:
nit / easily convinced otherwise: the implementers don't really care directly about SortByFieldName and in the future maybe there are other potential properties working together here.
What about `handleSortConfigurationChange`?